### PR TITLE
Feat/userId

### PR DESCRIPTION
**Pull request by bertacume**

### CODE REVIEW

File: src/script.js [hunk 1]
```diff
@@ -2,39 +2,66 @@
...
+    handleError(err, title, message);    
   });
+}
 
-});
+document.addEventListener('DOMContentLoaded', () => {
+  window.extAsyncInit = function () {
+    MessengerExtensions.getContext('{{{APP_ID}}}',
+      function success(thread_context) {
+        psid = thread_context.psid;
```
Comment (nandanrao) [inline src/script.js:59]: Handle race conditions! 

What if the vimeo loads and starts sending events before the messenger extensions getContext returns? psid will not be known! This should resolve a promise and the vimeo player should wait for that promise before loading and sending events...

File: src/script.js [hunk 2]
```diff
@@ -2,39 +2,66 @@
...
-    id,
-    width: 800
+    id: videoId,
+    responsive: true
   };
+
   const player = new Vimeo.Player('vimeoVideo', options);
 
   player.ready().then(() => {
     player.on('ended', data => handleEvent(data, 'ended'));
```
Comment (nandanrao) [inline src/script.js]: I suspect we should close the webview with MessengerExtensions.requestCloseBrowser when the user has finished the video, no?

**Review by nandanrao — CHANGES_REQUESTED** [review]

**Comment by nandanrao** [comment]
@bertacume -- did you have a chance to look at these comments? Let me know, thanks!

File: src/script.js (hunk 1 above)
Comment (bertacume) [reply src/script.js:59]: But I'm not setting the player until I receive the `thread_context`.

Comment (nandanrao) [reply src/script.js:59]: Hehehe, good point! :D

**Pull request merged**